app/src/main/java/org/schabi/newpipe/streams/OggFromWebMWriter.java

```diff
@@ -0,0 +1,434 @@
+        buffer.put((byte) 0x00);// version
+        buffer.put(packet_flag);// type
+
+        buffer.putLong(gran_pos);// granulate position
+
+        buffer.putInt(STREAM_ID);// bitstream serial number
+        buffer.putInt(sequence_count++);// page sequence number
+
+        buffer.putInt(0x00);// page checksum
+
+        buffer.put((byte) segment_table_size);// segment table
+        buffer.put(segment_table, 0, segment_table_size);// segment size
+
+        length += segment_table_size;
+
+        clearSegmentTable();// clear segment table for next header
+
+        int checksum_crc32 = calc_crc32(0x00, buffer.array(), length);
+
+        if (immediate_page != null) {
+            checksum_crc32 = calc_crc32(checksum_crc32, immediate_page, immediate_page.length);
+            buffer.putInt(HEADER_CHECKSUM_OFFSET, checksum_crc32);
+            segment_table_next_timestamp -= TIME_SCALE_NS;
+        }
+
+        return checksum_crc32;
+    }
+
+    @Nullable
+    private byte[] make_metadata() {
+        if ("A_OPUS".equals(webm_track.codecId)) {
+            return new byte[]{
+                    0x4F, 0x70, 0x75, 0x73, 0x54, 0x61, 0x67, 0x73,// "OpusTags" binary string
+                    0x07, 0x00, 0x00, 0x00,// writting application string size
+                    0x4E, 0x65, 0x77, 0x50, 0x69, 0x70, 0x65,// "NewPipe" binary string
+                    0x00, 0x00, 0x00, 0x00// additional tags count (zero means no tags)
+            };
+        } else if ("A_VORBIS".equals(webm_track.codecId)) {
+            return new byte[]{
+                    0x03,// ????????
+                    0x76, 0x6f, 0x72, 0x62, 0x69, 0x73,// "vorbis" binary string
+                    0x07, 0x00, 0x00, 0x00,// writting application string size
+                    0x4E, 0x65, 0x77, 0x50, 0x69, 0x70, 0x65,// "NewPipe" binary string
+                    0x01, 0x00, 0x00, 0x00,// additional tags count (zero means no tags)
+
+                    /*
+                        // whole file duration (not implemented)
+                        0x44,// tag string size
+                        0x55, 0x52, 0x41, 0x54, 0x49, 0x4F, 0x4E, 0x3D, 0x30, 0x30, 0x3A, 0x30, 0x30, 0x3A, 0x30,
+                        0x30, 0x2E, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30
+                     */
+                    0x0F,// tag string size
+                    0x00, 0x00, 0x00, 0x45, 0x4E, 0x43, 0x4F, 0x44, 0x45, 0x52, 0x3D,// "ENCODER=" binary string
+                    0x4E, 0x65, 0x77, 0x50, 0x69, 0x70, 0x65,// "NewPipe" binary string
+                    0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00// ????????
+            };
+        }
+
+        // not implemented for the desired codec
+        return null;
+    }
+
+    private void write(ByteBuffer buffer) throws IOException {
+        output.write(buffer.array(), 0, buffer.position());
+        buffer.position(0);
+    }
+
+    //<editor-fold defaultstate="collapsed" desc="WebM track handling">

```

Comment:
What are those comments for? Did I miss something before? There are some more `<editor>` comments below.